### PR TITLE
Fix the getCanonicalFileName in sys.ts and also check null value in f…

### PR DESCRIPTION
**Pull request by zhengbli**
…ile watcher call back

This PR fixes the `getCanonicalFileName` function used in `sys.ts` for file systems using case sensitive file names. Also added checking for `null` for file watcher callbacks as node sometimes will pass `null` as a parameter value.

### CODE REVIEW

File: src/compiler/sys.ts [hunk 1]
```diff
@@ -378,7 +378,11 @@ namespace ts {
                  */
                 function fileEventHandler(eventName: string, relativefileName: string, baseDirPath: Path) {
                     // When files are deleted from disk, the triggered "rename" event would have a relativefileName of "undefined"
-                    const filePath = relativefileName === undefined ? undefined : toPath(relativefileName, baseDirPath, getCanonicalPath);
+                    /* tslint:disable:no-null */
+                    const filePath = relativefileName === undefined || relativefileName === null
```
Comment (mhegazy) [inline src/compiler/sys.ts]: `typeof relativeFileName !== "string"`?

File: src/compiler/sys.ts [hunk 2]
```diff
@@ -378,7 +378,11 @@ namespace ts {
                  */
                 function fileEventHandler(eventName: string, relativefileName: string, baseDirPath: Path) {
                     // When files are deleted from disk, the triggered "rename" event would have a relativefileName of "undefined"
-                    const filePath = relativefileName === undefined ? undefined : toPath(relativefileName, baseDirPath, getCanonicalPath);
```
Comment (mhegazy) [inline src/compiler/sys.ts]: nit: relativeFileName

**Comment by mhegazy** [comment]
:+1:  with my minor comments.

File: src/compiler/sys.ts (hunk 1 above)
Comment (DanielRosenwasser) [inline src/compiler/sys.ts]: :+1:

**Pull request merged**